src/main/java/seedu/address/model/person/Room.java

```diff
@@ -21,15 +22,33 @@
      */
     public Room(String room) {
         requireNonNull(room);
-        checkArgument(isValidRoom(room), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidRoom(room), getMessageConstraints());
         value = room;
     }
 
     /**
      * Returns true if a given string is a valid Room number.
      */
     public static boolean isValidRoom(String test) {
-        return test.matches(VALIDATION_REGEX);
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        }
+        int room = Integer.parseInt(test);
+        boolean isRoomValid = room % 100 <= roomPref % 100 && room % 100 > 0;
+        boolean isFloorValid = Math.floorDiv(room, 100) <= Math.floorDiv(roomPref, 100)
+                && Math.floorDiv(room, 100) > 0;
+        return isFloorValid && isRoomValid;
+    }
+
+    public static String getMessageConstraints() {
+        int floorMax = Math.floorDiv(roomPref, 100);
+        int roomMax = roomPref % 100;
+        return MESSAGE_CONSTRAINTS + "\nFloor : 1 to " + String.valueOf(floorMax)
+                + "\nRoom No : 1 to " + String.valueOf(roomMax);
+    }
+
+    public static void setRoomPref(int pref) {
+        roomPref = pref;
```

Comment:
Move this method to `UserPrefs` as it is more related to it? E.g. `setGuiSettings` in `UserPrefs`

Reply:
`UserPrefs` is just a class to store the info of the preferences. It does not actually modify another class attributes